### PR TITLE
apply_storage and ether_storage update changes

### PR DESCRIPTION
**Pull request by AnastasiyaTarasova**
apply.rs apply_storage changes
ether_storage.rs update changes

### CODE REVIEW

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10735597230/artifacts/1900504774)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |  445556 |      0     |      0.342 |    0.00114244  |        0      |
| getSlot                           |  310393 |      0     |      0.353 |    0.000814339 |        0      |
| getBlocks                         |  101356 |      0     |      0.158 |    0.000652117 |        0      |
| getMultipleAccounts               |   95618 |      0     |      0.274 |    0.00145617  |        0.001  |
| getTransaction                    |   92383 |      0     |      0.37  |    0.00590351  |        0.004  |
| getBlock                          |   65657 |      0     |      0.224 |    0.00139249  |        0.001  |
| getLatestBlockhash                |   64332 |      0     |      0.363 |    0.00143933  |        0      |
| sendTransaction                   |   53766 |      0     |      0.379 |    0.00653324  |        0.003  |
| getBlockTime                      |   50851 |      0     |      0.198 |    0.000362648 |        0      |
| getBalance                        |   15389 |      0     |      0.108 |    0.000937163 |        0      |
| getSignatureStatuses              |    1456 |      0     |      0.026 |    0.000975962 |        0      |
| getVersion                        |     752 |      0     |      0.081 |    0.00114229  |        0      |
| isBlockhashValid                  |     710 |      0     |      0.024 |    0.00035493  |        0      |
| getSignaturesForAddress           |      67 |      0     |      0.003 |    0.00135821  |        0.001  |
| simulateTransaction               |      38 |      0     |      0.001 |    0.0005      |        0.0005 |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.00075     |        0.001  |
| getHealth                         |       3 |      0.001 |      0.004 |    0.003       |        0.004  |
| getTokenAccountBalance            |       3 |      0.001 |      0.001 |    0.001       |        0.001  |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |        0.001  |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |        0.001  |
| getClusterNodes                   |       1 |      0.001 |      0.001 |    0.001       |        0.001  |

</details>

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10740791929/artifacts/1901783450)

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10746391306/artifacts/1903455599)

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10773751796/artifacts/1909241986)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |  447337 |      0     |      0.233 |    0.00123665  |         0     |
| getSlot                           |  306658 |      0     |      0.263 |    0.000934177 |         0     |
| getBlocks                         |   99759 |      0     |      0.203 |    0.000687848 |         0     |
| getMultipleAccounts               |   96973 |      0     |      0.334 |    0.00167633  |         0.001 |
| getTransaction                    |   92628 |      0     |      0.659 |    0.00658311  |         0.004 |
| getBlock                          |   65876 |      0     |      0.067 |    0.00143844  |         0.001 |
| getLatestBlockhash                |   64492 |      0     |      0.242 |    0.00168069  |         0     |
| sendTransaction                   |   53850 |      0     |      0.561 |    0.00735742  |         0.004 |
| getBlockTime                      |   51512 |      0     |      0.091 |    0.000372088 |         0     |
| getBalance                        |   15416 |      0     |      0.103 |    0.000958809 |         0     |
| getSignatureStatuses              |    1501 |      0     |      0.037 |    0.00101532  |         0     |
| getVersion                        |     775 |      0     |      0.03  |    0.00115484  |         0     |
| isBlockhashValid                  |     732 |      0     |      0.02  |    0.000407104 |         0     |
| getSignaturesForAddress           |      65 |      0     |      0.017 |    0.0016      |         0.001 |
| simulateTransaction               |      38 |      0     |      0.001 |    0.000289474 |         0     |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.00065625  |         0.001 |
| getHealth                         |       3 |      0     |      0.004 |    0.00166667  |         0.001 |
| getTokenAccountBalance            |       3 |      0     |      0     |    0           |         0     |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |         0     |
| getClusterNodes                   |       1 |      0.002 |      0.002 |    0.002       |         0.002 |

</details>

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/10937023104/artifacts/1952076521)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |  446162 |      0     |      0.405 |    0.00171963  |         0.001 |
| getSlot                           |  310997 |      0     |      0.411 |    0.00120784  |         0.001 |
| getBlocks                         |  102420 |      0     |      0.164 |    0.000859402 |         0.001 |
| getMultipleAccounts               |   97092 |      0     |      0.372 |    0.00215399  |         0.001 |
| getTransaction                    |   92425 |      0     |      0.449 |    0.00758606  |         0.005 |
| getBlock                          |   66283 |      0     |      0.151 |    0.00188878  |         0.002 |
| getLatestBlockhash                |   64596 |      0     |      0.406 |    0.00194353  |         0.001 |
| sendTransaction                   |   53764 |      0     |      0.684 |    0.008699    |         0.005 |
| getBlockTime                      |   51505 |      0     |      0.116 |    0.000586312 |         0     |
| getBalance                        |   15376 |      0     |      0.154 |    0.00140485  |         0.001 |
| getSignatureStatuses              |    1486 |      0     |      0.066 |    0.00177725  |         0.001 |
| getVersion                        |     767 |      0     |      0.107 |    0.00204563  |         0.001 |
| isBlockhashValid                  |     725 |      0     |      0.034 |    0.00084     |         0     |
| getSignaturesForAddress           |      65 |      0.001 |      0.02  |    0.00303077  |         0.001 |
| simulateTransaction               |      38 |      0     |      0.001 |    0.000342105 |         0     |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.00090625  |         0.001 |
| getHealth                         |       3 |      0.001 |      0.013 |    0.00533333  |         0.002 |
| getTokenAccountBalance            |       3 |      0     |      0.001 |    0.000333333 |         0     |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |         0     |
| getClusterNodes                   |       1 |      0.014 |      0.014 |    0.014       |         0.014 |

</details>

File: evm_loader/lib/src/account_storage.rs [hunk 1]
```diff
@@ -1270,6 +1271,13 @@ impl<T: Rpc> SyncedAccountStorage for EmulatorAccountStorage<'_, T> {
                 .map_err(map_neon_error)?
                 .borrow_mut();
 
+            if storage_data.owner != system_program::ID
+                && storage_data.is_empty()
```
Comment (anton-lisanin) [inline evm_loader/lib/src/account_storage.rs]: `(storage_data.owner != system_program::ID) && storage_data.is_empty()` is almost never true.
For non existing accounts: `storage_data.owner == system_program::ID` and `storage_data.is_empty()`
For existing: `storage_data.owner != system_program::ID` and in 99.99% cases `storage_data` is not empty.

**Review by anton-lisanin — CHANGES_REQUESTED** [review]

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/11029850737/artifacts/1976091164)

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/11031941912/artifacts/1976618050)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |  446852 |      0     |      0.318 |    0.00129095  |        0      |
| getSlot                           |  294131 |      0     |      0.249 |    0.000824459 |        0      |
| getBlocks                         |   96419 |      0     |      0.157 |    0.000673778 |        0      |
| getMultipleAccounts               |   96268 |      0     |      0.456 |    0.00150964  |        0.001  |
| getTransaction                    |   92581 |      0     |      0.303 |    0.00544486  |        0.004  |
| getBlock                          |   63294 |      0     |      0.295 |    0.00145816  |        0.001  |
| getLatestBlockhash                |   63247 |      0     |      0.246 |    0.00127027  |        0      |
| sendTransaction                   |   53842 |      0     |      0.374 |    0.00610624  |        0.004  |
| getBlockTime                      |   51766 |      0     |      0.181 |    0.000374841 |        0      |
| getBalance                        |   15408 |      0     |      0.139 |    0.000987799 |        0      |
| getSignatureStatuses              |    1486 |      0     |      0.034 |    0.00102692  |        0      |
| getVersion                        |     767 |      0     |      0.033 |    0.00117601  |        0      |
| isBlockhashValid                  |     725 |      0     |      0.026 |    0.000395862 |        0      |
| getSignaturesForAddress           |      64 |      0     |      0.015 |    0.002125    |        0.001  |
| simulateTransaction               |      38 |      0     |      0.001 |    0.000315789 |        0      |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.000625    |        0.001  |
| getHealth                         |       3 |      0     |      0.034 |    0.0123333   |        0.003  |
| getTokenAccountBalance            |       3 |      0     |      0.001 |    0.000333333 |        0      |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |    0.0005      |        0.0005 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |        0      |
| getClusterNodes                   |       1 |      0.012 |      0.012 |    0.012       |        0.012  |

</details>

File: evm_loader/lib/src/account_storage.rs (hunk 1 above)
Comment (AnastasiyaTarasova) [reply evm_loader/lib/src/account_storage.rs]: Removed "storage_data.owner != system_program::ID && storage_data.is_empty()"   [2c20d0e](https://github.com/neonlabsorg/neon-evm/pull/541/commits/0aaaecaca1edf4333751ed2de27b3b9f0ceabfa5) - tests failed 

Removed only ""storage_data.owner != system_program::ID"  [484fb43](https://github.com/neonlabsorg/neon-evm/pull/541/commits/50aba8f2bb8d2dd14181b9b92b3522dd399bc106) - tests passed

**Review by anton-lisanin — APPROVED** [review]

**Pull request merged**